src/nvc.rs

```diff
@@ -2,6 +2,100 @@ use instruction::*;
 use interconnect::*;
 
 use std::collections::HashSet;
+use std::fmt;
+
+#[derive(Copy, Clone)]
+pub struct NvcCacheEntry {
+    pub tag: u32,
+    pub base_addr: u32,
+    pub subblock_valid: [bool; 2],
+}
+
+impl NvcCacheEntry {
+    fn new() -> NvcCacheEntry {
+        NvcCacheEntry {
+          tag: 0,
+          base_addr: 0,
+          subblock_valid: [false; 2],
+        }
+    }
+}
+
+impl fmt::Display for NvcCacheEntry {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "base_addr: 0x{:08x}, tag: 0x{:08x}, sb1 valid: {:5}, sb2 valid: {:5}", self.base_addr, self.tag, self.subblock_valid[0], self.subblock_valid[1])
+    }
+}
+
+pub struct NvcCache {
+    hits: usize,
+    misses: usize,
+    cache_entries: [NvcCacheEntry; 128],
+}
+
+impl NvcCache {
+    pub fn new() -> NvcCache {
+        NvcCache {
+          hits: 0,
+          misses: 0,
+          cache_entries: [NvcCacheEntry::new(); 128],
+        }
+    }
+
+    pub fn clear_entries(&mut self, entry_start: usize, count: usize) {
+        if entry_start >= 128 {
+            return;
+        }
+        let mut entry_count = count;
```

Comment:
nit: if you want to, you can declare `count` as `mut count` (or perhaps as `mut entry_count`), which might be slightly clearer than making a new binding just to mutate it.